src/plcc/parse_spec/parse_syntactic_spec/parse_syntactic_spec.py

```diff
@@ -0,0 +1,133 @@
+
+
+def parse_right(line: Line, rhs: str) -> List[Symbol]:
+    total = []
+    if not rhs:
+        return total
+    symbols = rhs.split(" ")
+    for symbol in symbols:
+        if symbol == "":
+            continue
+        if symbol.startswith("#") or symbol is None:
+            return total
+        total.append(parse_symbol(symbol))
+    return total
+
+
+def parse_symbol(symbol: str) -> Symbol:
+    capturing = re.match(r"<(?P<name>\S*)>(?P<alt_name>\S+)?", symbol)
+
+    if capturing:
+        name = capturing["name"]
+        alt_name = capturing["alt_name"]
+        return parse_capturing(name, alt_name)
+    return Terminal(symbol)
+
+
+def parse_capturing(name: str, alt_name: str) -> CapturingSymbol:
+    terminal = re.match(r"[A-Z_]+", name)
+
+    if alt_name is not None:
+        alt_name = alt_name.strip(":")
+
+    if terminal:
+        return CapturingTerminal(name, alt_name)
+    return RhsNonTerminal(name, alt_name)
+
+
+def is_skippable(line: Line) -> bool:
+    empty = re.match(r"^\s*$", line.string)
+    return empty is not None or line.string.startswith("#")
+
+
+def build_matches(
+    line: str,
+) -> tuple[Match[str] | None, Match[str] | None, Match[str] | None]:
+
+    standard = re.match(r"^\s*(?P<lhs><\S+>(?::\S+)?)\s*::=(?:\s(?P<rhs>.*))?", line)
+    separated = re.match(
+        r"^\s*(?P<lhs><\S+>(?::\S+)?)\s*\*\*=\s(?P<rhs>.*)(?P<separator>\+.*)",
+        line,
+    )
+    repeating = re.match(r"^\s*(?P<lhs><\S+>(?::\S+)?)\s*\*\*=\s(?P<rhs>.*)", line)
+
+    return standard, separated, repeating
+
+
+def parse_syntactic_rule(line: Line) -> SyntacticRule:
+
+    standard, separated, repeating = build_matches(line.string)
+
+    if standard:
+        return StandardSyntacticRule(
+            line, parse_left(line, standard["lhs"]), parse_right(line, standard["rhs"])
+        )
+
+    elif separated:
+        return parse_repeating(
+            line,
+            separated["lhs"],
+            separated["rhs"],
+            separated["separator"].strip("+").strip(" "),
+        )
+
+    elif repeating:
+        return parse_repeating(line, repeating["lhs"], repeating["rhs"], None)
+
+    raise MalformedBNFError(line)
+
+
+def parse_repeating(
+    line: Line, left: str, right: str, separator: str | None
+) -> RepeatingSyntacticRule:
+    if separator:
+        return RepeatingSyntacticRule(
+            line,
+            parse_left(line, left),
+            parse_right(line, right),
+            Terminal(separator.strip("+")),
+        )
+    else:
+        return RepeatingSyntacticRule(
+            line, parse_left(line, left), parse_right(line, right), None
+        )
+
+
+def parse_syntactic_spec(lines: List[Line | Divider] | None) -> SyntacticSpec:
+    s = SyntacticSpec()
+    if not lines:
+        return s
+    for line in lines:
+        if isinstance(line, Divider) or is_skippable(line):
+            continue
+        s.append(parse_syntactic_rule(line))
+    return s
```

Comment:
I would recommend using a class-based approach and having this method outside of the class, so it can be called from outside, and have this method instantiate a helper class to handle the logic. Also, try to use camelCase for the function and variable names above, you can leave this function name as snake_case, and use _ before the names of the helper classes you define, below is a brief example of the class-based approach:
```
class SyntacticParserHelper
...
parser = SyntacticParserHelper(line)
```